language/Type-Classes.md

```diff
@@ -137,36 +137,128 @@ See also the section in [PureScript by Example](https://book.purescript.org/chap
 
-derive instance eqPerson :: Eq Person
-derive instance ordPerson :: Ord Person
+tenPoints :: Points
+tenPoints = (Points 4) + (Points 6)
 ```
-Currently, the following type classes can be derived by the compiler:
 
+That `derive` line replaced all this code!:
+
+```purs
+-- No need to write this
+instance semiringPoints :: Semiring Points where
+  zero = Points 0
+  add (Points a) (Points b) = Points (a + b)
+  mul (Points a) (Points b) = Points (a * b)
+  one = Points 1
+```
+
+### Classes with built-in compiler support
+
+Some classes have special built-in compiler support, and instances can be derived from all types, not just `newtype`s.
+
+For example, if you you'd like to be able to remove duplicates from an array of an ADT using `nub`, you need an `Eq` and `Ord` instance. Rather than writing these manually, let the compiler do the work.
+
+```purs
+import Data.Array (nub)
+
+data MyADT
+  = Some
+  | Arbitrary Int
+  | Contents Number String
+
+derive instance eqMyADT :: Eq MyADT
+derive instance ordMyADT :: Ord MyADT
+
+nub [Some, Arbitrary 1, Some, Some] == [Some, Arbitrary 1]
+```
+
+Currently, the following type classes can be derived by the compiler:
 - [Data.Generic.Rep (class Generic)](https://pursuit.purescript.org/packages/purescript-generics-rep/docs/Data.Generic.Rep#t:Generic)
 - [Data.Eq (class Eq)](https://pursuit.purescript.org/packages/purescript-prelude/docs/Data.Eq#t:Eq)
 - [Data.Ord (class Ord)](https://pursuit.purescript.org/packages/purescript-prelude/docs/Data.Ord#t:Ord)
 - [Data.Functor (class Functor)](https://pursuit.purescript.org/packages/purescript-prelude/docs/Data.Functor#t:Functor)
 - [Data.Newtype (class Newtype)](https://pursuit.purescript.org/packages/purescript-newtype/docs/Data.Newtype#t:Newtype)
 
-Note that `derive instance` is not the only mechanism for allowing you to avoid writing out boilerplate type class instance code. Many type classes not listed here can be derived through other means, such as via a Generic instance. For example, here's how to create a `Show` instance for `Person` via `genericShow`:
+Note that with our earlier `Points` `newtype`, we can use either of these options to derive an `Eq` instance. They are equivalent in this case.
+
+```purs
+derive instance eqPoints :: Eq Points
+derive newtype instance eqPoints :: Eq Points
+```
+
+### Deriving from `Generic`
+
+The compiler's built-in support for `Generic` unlocks convenient deriving for many other classes not listed above.
+
+For example, if we wanted a `Show` instance for `MyADT`, it might seem like we're out of luck because neither `MyADT` is a `newtype` nor is `Show` listed as a class with built-in compiler support.
+
+But we _can_ use `genricShow`, which works with _any_ type that has a `Generic` instance. And recall that the compiler has built-in support for deriving a `Generic` instance for any type (including the `MyADT` type). We put all those pieces together like so:
```

Comment:
```suggestion
But we _can_ use `genericShow`, which works with _any_ type that has a `Generic` instance. And recall that the compiler has built-in support for deriving a `Generic` instance for any type (including the `MyADT` type). We put all those pieces together like so:
```

This might be the better place to demonstrate the `Show` differences between newtype deriving and generic deriving, actually, rather than earlier. I believe that the generic version of `Points` will include the constructor and the newtype derived version will not.

Reply:
Geeze. I should read all the comments before commenting myself.